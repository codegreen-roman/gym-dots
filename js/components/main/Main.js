import React from 'react'
import moment from 'moment'

import { Header } from '../common/header/Header.connect'
import { Footer } from '../common/footer/Footer.connect'
import { Route, Switch, NavLink } from 'react-router-dom'
import { Activity } from '../views/activity/Activity'
import { Overlay } from '../common/overlay/Overlay.connect'
import { Countdown } from '../common/overlay/Countdown'
import { mainStyles } from './Main.glamor'
import { Welcome } from '../views/welcome/Welcome.connect'

export const Main = () => {
    const dateStr = moment().format('dddd, MMM Do')
    const headerTitle = 'Start Workout'

    return (
        <section className='app-container' {...mainStyles}>
            <Switch>
                <Route
                    path='/in'
                    component={() => {
                        return (
                            <main>
                                <Header
                                    dateStr={dateStr}
                                    subTitle={headerTitle}
                                />
                                <Switch>
                                    <Route
                                        path='/activity'
                                        component={Activity}
                                    />
                                    <Route
                                        path='/'
                                        component={() =>
                                            <NavLink to='/activity'>
                                                Start training
                                            </NavLink>}
                                    />
                                </Switch>
                                <Footer />
                                <Overlay>
                                    <Countdown />
                                </Overlay>
                            </main>
                        )
                    }}
                />
                <Route path='/' component={() => <Welcome dateStr={dateStr} />}
                />
                <Route component={() => <section>404 Not Found</section>} />
            </Switch>
        </section>
    )
}
