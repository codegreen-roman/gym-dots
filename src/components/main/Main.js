import React from 'react'
import moment from 'moment'

import { Header } from '@components/common/header/Header.connect'
import { Footer } from '@components/common/footer/Footer.connect'
import { Route, Switch, NavLink } from 'react-router-dom'
import { Activity } from '../views/activity/Activity'
import { Overlay } from '@components/common/overlay/Overlay.connect'
import { Countdown } from '@components/common/overlay/Countdown'
import { StyledAppContainer, StyledMain } from './Main.glamor'
import { Welcome } from '../views/welcome/Welcome.connect'

export const Main = () => {
    const dateStr = moment().format('dddd, MMM Do')
    const headerTitle = 'Start Workout'

    return (
        <StyledAppContainer>
            <Switch>
                <Route
                    path='/in'
                    component={({ match }) => {
                        return (
                            <StyledMain>
                                <Header
                                    dateStr={dateStr}
                                    subTitle={headerTitle}
                                />
                                <Switch>
                                    <Route
                                        path={`${match.url}/activity`}
                                        component={Activity}
                                    />
                                    <Route
                                        path={`${match.url}/`}
                                        component={() =>
                                            <NavLink to={`${match.url}/activity`}>
                                                Start training
                                            </NavLink>}
                                    />
                                </Switch>
                                <Footer />
                                <Overlay>
                                    <Countdown />
                                </Overlay>
                            </StyledMain>
                        )
                    }}
                />

                <Route path='/' component={() => <Welcome dateStr={dateStr} />}
                />
                <Route component={() => <section>404 Not Found</section>} />
            </Switch>
        </StyledAppContainer>
    )
}
