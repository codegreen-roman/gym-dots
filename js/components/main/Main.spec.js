import React from 'react'
import configureMockStore from 'redux-mock-store'
import { shallow } from 'enzyme'

import { Main } from './Main'

const setup = () => {
    const store = configureMockStore()()
    const wrapper = shallow(<Main store={store} />)

    return {
        store,
        wrapper
    }
}

describe('App', () => {
    test('renders without crashing', () => {
        const { wrapper } = setup()
        expect(wrapper).toMatchSnapshot()
    })
})
