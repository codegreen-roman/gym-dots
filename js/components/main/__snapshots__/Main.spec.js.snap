// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<section
  className="app-container"
  data-css-wivn7t=""
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/welcome"
    />
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</section>
`;
